src/NuGet.Core/NuGet.Build.Tasks/GenerateTemporaryTargetAssembly2.cs

```diff
@@ -0,0 +1,523 @@
+
+namespace Microsoft.Build.Tasks.Windows
+{
+    #region GenerateTemporaryTargetAssembly2 Task class
+
+    /// <summary>
+    ///   This task is used to generate a temporary target assembly. It generates
+    ///   a temporary project file and then compile it.
+    /// 
+    ///   The generated project file is based on current project file, with below
+    ///   modification:
+    /// 
+    ///       A:  Add the generated code files (.g.cs) to Compile Item list.
+    ///       B:  Replace Reference Item list with ReferenctPath item list.
+    ///           So that it doesn't need to rerun time-consuming tatk 
+    ///           ResolveAssemblyReference (RAR) again.
+    /// 
+    /// </summary>
+    public sealed class GenerateTemporaryTargetAssembly2 : Task
+    {
+        //------------------------------------------------------
+        //
+        //  Constructors
+        //
+        //------------------------------------------------------
+
+        #region Constructors
+
+        /// <summary>
+        /// Constrcutor
+        /// </summary>
+        public GenerateTemporaryTargetAssembly2()
+        {
+        }
+
+        #endregion Constructors
+
+        //------------------------------------------------------
+        //
+        //  Public Methods
+        //
+        //------------------------------------------------------
+
+        #region Public Methods
+
+        /// <summary>
+        /// ITask Execute method
+        /// </summary>
+        /// <returns></returns>
+        public override bool Execute()
+        {
+            bool retValue = true;
+
+            // Verification
+            try
+            {
+                XmlDocument xmlProjectDoc = null;
+
+                xmlProjectDoc = new XmlDocument();
+                xmlProjectDoc.Load(CurrentProject);
+
+                //
+                // remove all the WinFX specific item lists
+                // ApplicationDefinition, Page, MarkupResource and Resource
+                //
+
+                RemoveItemsByName(xmlProjectDoc, APPDEFNAME);
+                RemoveItemsByName(xmlProjectDoc, PAGENAME);
+                RemoveItemsByName(xmlProjectDoc, MARKUPRESOURCENAME);
+                RemoveItemsByName(xmlProjectDoc, RESOURCENAME);
+
+                // Replace the Reference Item list with ReferencePath
+
+                RemoveItemsByName(xmlProjectDoc, REFERENCETYPENAME);
+                AddNewItems(xmlProjectDoc, ReferencePathTypeName, ReferencePath);
+
+                // Add GeneratedCodeFiles to Compile item list.
+                AddNewItems(xmlProjectDoc, CompileTypeName, GeneratedCodeFiles);
+
+
+                // Create a random file name
+                // This can fix the problem of project cache in VS.NET environment.
+                //
+
+                // GetRandomFileName( ) could return any possible file name and extension, but
+                // some file externsion has special meaning in MSBUILD system, such as a ".sln"
```

Comment:
typo: `externsion`

(even if you copied it from src)

Reply:
Fixing in next iteration. Yes, was copied.